compiler/sem/semfold.nim

```diff
@@ -562,30 +566,33 @@ proc getConstExpr(m: PSym, n: PNode; idgen: IdGenerator; g: ModuleGraph): PNode
       of mBuildCPU: result = newStrNodeT(platform.CPU[g.config.target.hostCPU].name.toLowerAscii, n, g)
       of mAppType: result = getAppType(n, g)
       of mIntDefine:
-        if isDefined(g.config, s.name.s):
-          try:
-            result = newIntNodeT(toInt128(g.config.symbols[s.name.s].parseInt), n, idgen, g)
-          except ValueError:
-            g.config.localReport reportStr(
-              rsemInvalidIntdefine, g.config.symbols[s.name.s])
-
-        else:
-          result = copyTree(s.ast)
+        result =
```

Comment:
Given the similarity of the three branches, I think it would make sense to consolidate them into a single one, like so:
```nim
of mIntDefine, mStrDefine, mBoolDefine:
  result =
    if isDefined(g.config, s.name.s):
      let define = g.config.symbols[s.name.s]
      case s.magic
      of mIntDefine: ...
```
That's one more level of nesting, however, which is a bit meh. No change is expected from my side.

As an aside (nothing for this PR), resolving the `define` constants during folding feels off. Replacing the constant's value in `semNormalizedConst` would be better, I think. Doing so, would make retrieving the original value impossible, although I don't think that'd would be problematic.

Reply:
Yeah, the define registration feels a bit off here. I think one question to answer is what do these defines really mean? Are they truly the equivalent of the user writing something like:

```nim
const foo = if defined(foo): getDefine(foo).parseInt else: 10
```

If so, then I think it's fair to say that the if expression is a subroutine (template) and so:
- we set the value in `semNormalizedConst`
- remember the default via template expansion (or a specialized equivalent)

Reply:
I updated the code style-wise; leaving the bigger what is a an `intdefine` etc as out of scope for this PR.

Won't resolve conversation for visilbity.